test/hotspot/jtreg/compiler/vectorapi/TestReverseByteTransforms.java

```diff
@@ -0,0 +1,306 @@
+    }
+
+    @Run(test = {"test_reversebytes_long_transform1"}, mode = RunMode.STANDALONE)
+    public void kernel_test_reversebytes_long_transform1() {
+        for (int i = 0; i < ITERS; i++) {
+            test_reversebytes_long_transform1(lout, linp);
+        }
+    }
+
+    @Test
+    @IR(applyIfCPUFeatureOr={"sve", "true", "simd", "true", "avx2", "true"}, counts = {"ReverseBytesV" , " > 0 "})
+    public void test_reversebytes_long_transform2(long[] lout, long[] linp) {
+        VectorMask<Long> mask = VectorMask.fromLong(LSPECIES, 3);
+        for (int i = 0; i < LSPECIES.loopBound(linp.length); i+=LSPECIES.length()) {
+            LongVector.fromArray(LSPECIES, linp, i)
+                     .lanewise(VectorOperators.REVERSE_BYTES)
+                     .lanewise(VectorOperators.REVERSE_BYTES, mask)
+                     .intoArray(lout, i);
+        }
+    }
+
+    @Run(test = {"test_reversebytes_long_transform2"}, mode = RunMode.STANDALONE)
+    public void kernel_test_reversebytes_long_transform2() {
+        for (int i = 0; i < ITERS; i++) {
+            test_reversebytes_long_transform2(lout, linp);
+        }
+    }
+
+    @Test
+    @IR(applyIfCPUFeatureOr={"sve", "true", "simd", "true", "avx2", "true"}, counts = {"ReverseBytesV" , " > 0 "})
+    public void test_reversebytes_long_transform3(long[] lout, long[] linp) {
+        VectorMask<Long> mask1 = VectorMask.fromLong(LSPECIES, 3);
+        VectorMask<Long> mask2 = VectorMask.fromLong(LSPECIES, 3);
+        for (int i = 0; i < LSPECIES.loopBound(linp.length); i+=LSPECIES.length()) {
+            LongVector.fromArray(LSPECIES, linp, i)
+                     .lanewise(VectorOperators.REVERSE_BYTES, mask1)
+                     .lanewise(VectorOperators.REVERSE_BYTES, mask2)
+                     .intoArray(lout, i);
+        }
+    }
+
+    @Run(test = {"test_reversebytes_long_transform3"}, mode = RunMode.STANDALONE)
+    public void kernel_test_reversebytes_long_transform3() {
+        for (int i = 0; i < ITERS; i++) {
+            test_reversebytes_long_transform3(lout, linp);
+        }
+    }
+
+    @Test
+    @IR(applyIfCPUFeatureOr={"sve", "true", "simd", "true", "avx2", "true"}, failOn = {"ReverseBytesV" , " > 0 "})
+    public void test_reversebytes_long_transform4(long[] lout, long[] linp) {
+        for (int i = 0; i < LSPECIES.loopBound(linp.length); i+=LSPECIES.length()) {
+            LongVector.fromArray(LSPECIES, linp, i)
+                     .lanewise(VectorOperators.REVERSE_BYTES)
+                     .lanewise(VectorOperators.REVERSE_BYTES)
+                     .intoArray(lout, i);
+        }
+    }
+
+    @Run(test = {"test_reversebytes_long_transform4"}, mode = RunMode.STANDALONE)
+    public void kernel_test_reversebytes_long_transform4() {
+        for (int i = 0; i < ITERS; i++) {
+            test_reversebytes_long_transform4(lout, linp);
+        }
+    }
+
+    @Test
+    @IR(applyIfCPUFeature={"sve", "true"}, failOn = {"ReverseBytesV" , " > 0 "})
```

Comment:
Can we test the IR with x86 avx-512 predicated feature instead of "sve"? SVE is different that we also need to specify "-XX:UseSVE=1" to make sure the predicated feature enabled.

Reply:
IR check is not applicable for AVX512,  test point is added to cover transform for SVE which supports direct predicated vector instruction. Feature checks are much more strict and "sve" feature will be available only if UseSVE is set to 1.

Thanks, removed redundant SVE check from IR annotations from other places.

Reply:
Thanks for updating the tests! I ran the test in our internal testing system,  and unfortunately, this case will fail with "-XX:UseSVE=0" as expected. The reason is actually like what I said above. So to fix this, could you please:

1. Limit the whole test on aarch64 os systems by adding "`@requires vm.cpu.features ~= ".*simd.*"`" before the test. And limit the single IR test by adding "`applyIf={"UseSVE", ">0"}`" .

OR

2. Fix the IR framework, to make "applyIf" and "applyIfCPUFeature" can co-work with each other.

WDYT?

Reply:
I am not clear, if UseSVE=0 then why does "sve" feature getting populated during VM initialization?
Entire handling of applyIfCPUFeature* is based on white box CPU feature API which queries the feature list populated during VM initialization, we are not directly queries the target features seen in /proc/cpuinfo.

Reply:
Yeah, that's the difference from X86. `UseSVE` is a vm flag which lets user to choose whether use the sve feature or not. And the sve cpu feature is still there that will not been influenced by the VM flag.

Reply:
I think we should not be enabling sve feature if user explicitly passes VM flag UseSVE = 0, may be it should be fixed separately.
  
Option 1 seems like a good interim solution to me, will update the patch.

Reply:
Hi @XiaohongGong , kindly check and approve.

Reply:
Tests pass, and thanks for the updating!